bootstore/src/schemes/v0/state_initial_member.rs

```diff
@@ -0,0 +1,411 @@
+                        to.clone(),
+                        metadata.request_id,
+                        &self.pkg,
+                        &mut self.distributed_shares,
+                        &shares,
+                    );
+
+                    if persist_once {
+                        persist = true;
+                    }
+                    envelopes.extend(envelopes_once);
+                }
+            }
+            Err(_) => {
+                envelopes = self.resolve_learn_requests_with_error(
+                    Error::FailedToDecryptShares,
+                );
+            }
+        }
+
+        (persist, envelopes)
+    }
+
+    /// An error prevented successful resolution of learn requests
+    pub fn resolve_learn_requests_with_error(
+        &mut self,
+        error: Error,
+    ) -> Vec<Envelope> {
+        let mut envelopes =
+            Vec::with_capacity(self.pending_learn_requests.len());
+        while let Some((to, metadata)) = self.pending_learn_requests.pop_first()
+        {
+            let msg = Response {
+                request_id: metadata.request_id,
+                type_: error.into(),
+            };
+            envelopes.push(Envelope { to, msg: msg.into() })
+        }
+        envelopes
+    }
+}
+
+impl StateHandler for InitialMemberState {
+    fn handle_request(
+        mut self,
+        common: &mut FsmCommonData,
+        from: Baseboard,
+        request_id: Uuid,
+        request: RequestType,
+    ) -> (State, Output) {
+        use RequestType::*;
+        let output = match request {
+            Init(new_pkg) => {
+                if new_pkg == self.pkg {
+                    // Idempotent response given same pkg
+                    Output::respond(from, request_id, ResponseType::InitAck)
+                } else {
+                    let rack_uuid = self.pkg.rack_uuid;
+                    Output::respond(
+                        from,
+                        request_id,
+                        Error::AlreadyInitialized { rack_uuid }.into(),
+                    )
+                }
+            }
+            GetShare { rack_uuid } => {
+                if rack_uuid != self.pkg.rack_uuid {
+                    Output::respond(
+                        from,
+                        request_id,
+                        Error::RackUuidMismatch {
+                            expected: self.pkg.rack_uuid,
+                            got: rack_uuid,
+                        }
+                        .into(),
+                    )
+                } else {
+                    Output::respond(
+                        from,
+                        request_id,
+                        ResponseType::Share(self.pkg.share.clone()),
+                    )
+                }
+            }
+            Learn => {
+                let expiry =
+                    common.clock + common.config.rack_secret_request_timeout;
+                let output = common.rack_secret_state.load(
```

Comment:
Would it make sense to pass in `common` here, then remove `common.peers`, `common.id`, and `expiry` (which is computable from `common`)?

Same question for passing in `&self.pkg` instead of a bunch of pieces.

Reply:
No longer relevant as of [98ada27](https://github.com/oxidecomputer/omicron/pull/3385/commits/98ada27b53d4429463cde94cc91bedec0bc4ef98)